lib/resty/core/output.lua

```diff
@@ -0,0 +1,37 @@
+local ffi = require "ffi"
+local base = require "resty.core.base"
+local FFI_OK = base.FFI_OK
+local FFI_ERROR = base.FFI_ERROR
+local C = ffi.C
+
+local _M = {
+    version = base.version
+}
+
+ffi.cdef[[
+        int ngx_http_lua_ffi_write(ngx_http_request_t *r,
+                const char* str,
+                size_t offset,
+                size_t len);
```

Comment:
There is no point in putting each parameter in a separate line. You are far away from 80 columns.